erts/emulator/beam/erl_term.h

```diff
@@ -409,7 +409,12 @@ _ET_DECLARE_CHECKED(Eterm,bignum_header_neg,Eterm)
 #define _unchecked_bignum_header_arity(x)	_unchecked_header_arity((x))
 _ET_DECLARE_CHECKED(Uint,bignum_header_arity,Eterm)
 #define bignum_header_arity(x)	_ET_APPLY(bignum_header_arity,(x))
-#define BIG_ARITY_MAX		((1 << 19)-1)
+
+#if defined(ARCH_64)
+#  define BIG_ARITY_MAX		((1 << 16)-1)
+#else
+#  define BIG_ARITY_MAX		((1 << 17)-1)
+#endif
```

Comment:
What's the reason for this restriction? The commit doesn't say.

Reply:
It's the same in bytes though, as these values are in words (pointer-sized integers).

Reply:
We wanted to decrease the maximum execution time for a single arithmetic operation. We chose a new limit that we hope don't affect real applications of large integer arithmetic, but that will produce a `system_limit` exception faster for accidental use of large integers.